apps/api/src/app/events/usecases/trigger-event/message-filter.matcher.ts

```diff
@@ -145,9 +165,44 @@ async function getWebhookResponse(i, variables: IFilterVariables): Promise<Recor
   }
 }
 
-async function processFilter(variables: IFilterVariables, i) {
+async function buildPayload(variables: IFilterVariables, configuration: IMessageFilterConfiguration) {
+  const payload: Partial<{
+    subscriber: SubscriberEntity;
+    payload: Record<string, unknown>;
+    hmac: string;
+    identifier: string;
+  }> = {};
+
+  if (variables.subscriber) {
+    payload.subscriber = variables.subscriber;
+  } else {
+    payload.subscriber = await configuration.subscriberRepository.findOne({
+      _id: configuration.command.subscriberId,
+      _environmentId: configuration.command.environmentId,
+    });
+  }
+
+  if (variables.payload) {
+    payload.payload = variables.payload;
+  }
+
+  const environment = await configuration.environmentRepository.findOne({
+    _id: configuration.command.environmentId,
+    _organizationId: configuration.command.organizationId,
+  });
+
+  payload.hmac = createHmac('sha256', environment.apiKeys[0].key)
```

Comment:
Would suggest to create a custom header to include this, you can look at GitHub implementation for best practices: https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks#validating-payloads-from-github

Reply:
https://github.com/novuhq/novu/pull/2348/commits/63ae9a615b15ca4f3f11c71a9a7f216d9a37fe12